### PR TITLE
Added a Dockerfile and created a Docker compose YAML file

### PR DESCRIPTION
**Pull request by SimardeepSingh-zsh**
To build and run LiteLLM application in a Docker container, use the following commands:
# Build the Docker image
docker-compose build

# Run the Docker container
docker-compose up

This will start your application in a Docker container. You can access it using the specified port (e.g., http://localhost:8000/).

### CODE REVIEW

**Comment by ghost** [comment]
@SimardeepSingh-zsh added comments - i'm unclear on what the purpose of this dockerfile is. What in main.py is it supposed to run?

**Comment by SimardeepSingh-zsh** [comment]
Hi Krrish,

I understand your question about the purpose of the Dockerfile and what `main.py` is meant to run in our LiteLLM project. Let me break it down for you:

1. **Dockerfile**: The Dockerfile is a set of instructions that defines how to create a Docker image of our LiteLLM application. Docker is a containerization platform that allows us to package our application and its dependencies into a standardized container, ensuring consistency across different environments. In this case, our Dockerfile specifies that we should:
   - Use the official Python 3.8 image as a base image.
   - Set the working directory inside the container to `/app`.
   - Copy the `requirements.txt` file into the container and install the required Python packages using `pip`.
   - Copy the entire content of our application into the container.
   - Define that the command to run when the container starts is `python your_main_script.py`. This means it will execute `your_main_script.py` (which should be replaced with the actual name of your main script, which is `main.py` in our case).

2. **main.py**: `main.py` is the main script of our LiteLLM application. It serves as the entry point for our application. When the Docker container is started, it runs the code within `main.py`. This script is responsible for orchestrating the various functionalities of LiteLLM, such as interacting with language models, handling user requests, and generating responses.

In summary, the Dockerfile is used to create a containerized environment for our LiteLLM application, and `main.py` is the script that gets executed when the container starts. Inside `main.py`, we have the logic that drives our LiteLLM application and provides its core functionality.

If you have any more specific questions, please feel free to let me know. 

Best regards,
Simardeep Singh

**Comment by ghost** [comment]
closing based on conversation via DMs
